numba_dpex/core/kernel_interface/func.py

```diff
@@ -103,6 +108,28 @@ def __init__(self, pyfunc, debug=False, enable_cache=True):
             self._cache = NullCache()
         self._cache_hits = 0
 
+    def _create_func_hash(self):
+        """Creates a tuple of sha256 hashes out of code and variable bytes extracted from the compiled funtion."""
```

Comment:
Instead of duplicating the function in both dispatcher and func, how about keeping it in caching.py and calling it in both places?

Reply:
Moved to separate caching_utils.py